keras_nlp/src/models/video_swin/video_swin_layers.py

```diff
@@ -0,0 +1,1015 @@
+        height (int): Height of the video frames.
+        width (int): Width of the video frames.
+        window_size (tuple[int]): Size of the sliding window in each dimension
+            (depth, height, width).
+        shift_size (tuple[int]): Size of the shifting step in each dimension
+            (depth, height, width).
+
+    Returns:
+        A tensor of shape (batch_size, num_windows, num_windows), where:
+            - batch_size: Assumed to be 1 in this function.
+            - num_windows: Total number of windows covering the entire input based on
+                the formula:
+                    (depth - window_size[0]) // shift_size[0] + 1) *
+                    (height - window_size[1]) // shift_size[1] + 1) *
+                    (width - window_size[2]) // shift_size[2] + 1)
+        Each element (attn_mask[i, j]) represents the attention weight between
+        window i and window j. A value of -100.0 indicates high negative attention
+        (preventing information flow), 0.0 indicates no mask effect.
+    """
+
+    img_mask = np.zeros((1, depth, height, width, 1))
+    cnt = 0
+    for d in (
+        slice(-window_size[0]),
+        slice(-window_size[0], -shift_size[0]),
+        slice(-shift_size[0], None),
+    ):
+        for h in (
+            slice(-window_size[1]),
+            slice(-window_size[1], -shift_size[1]),
+            slice(-shift_size[1], None),
+        ):
+            for w in (
+                slice(-window_size[2]),
+                slice(-window_size[2], -shift_size[2]),
+                slice(-shift_size[2], None),
+            ):
+                img_mask[:, d, h, w, :] = cnt
+                cnt = cnt + 1
+    mask_windows = window_partition(img_mask, window_size)
+    mask_windows = ops.squeeze(mask_windows, axis=-1)
+    attn_mask = ops.expand_dims(mask_windows, axis=1) - ops.expand_dims(
+        mask_windows, axis=2
+    )
+    attn_mask = ops.where(attn_mask != 0, -100.0, attn_mask)
+    attn_mask = ops.where(attn_mask == 0, 0.0, attn_mask)
+    return attn_mask
+
+
+class MLP(keras.layers.Layer):
+    """A Multilayer perceptron(MLP) layer.
+
+    Args:
+        hidden_dim (int): The number of units in the hidden layer.
+        output_dim (int): The number of units in the output layer.
+        drop_rate  (float): Float between 0 and 1. Fraction of the
+            input units to drop.
+        activation (str): Activation to use in the hidden layers.
+            Default is `"gelu"`.
+
+    References:
+        - [Video Swin Transformer](https://arxiv.org/abs/2106.13230)
+        - [Video Swin Transformer GitHub](https://github.com/SwinTransformer/Video-Swin-Transformer)
+    """
+
+    def __init__(
+        self, hidden_dim, output_dim, drop_rate=0.0, activation="gelu", **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.output_dim = output_dim
+        self.hidden_dim = hidden_dim
+        self._activation_identifier = activation
```

Comment:
I see this in a few places, we should not do it.

in init
`self.activation = keras.activations.get(activation)`

in get_config
`"activation": keras.activations.serialize(self.activation),`